### PR TITLE
Adding example functions for more DB operations

### LINKED ISSUE #37: Extend example application to include all supported DynamoDB operations.
The example Golang application only has read operations, it should include all supported operations.

- [ ] BatchGetItem
- [ ] BatchWriteItem
- [ ] DeleteItem
- [ ] GetItem
- [ ] PutItem
- [ ] Query
- [ ] Scan
- [ ] UpdateItem

### PR DESCRIPTION
**Pull request by gushob21**
Fixes #37 

> It's a good idea to open an issue first for discussion.

- [x] Tests pass
- [x] Appropriate changes to README are included in PR

### CODE REVIEW

File: examples/golang/main.go [hunk 1]
```diff
@@ -229,13 +249,434 @@ func getProductsByCategory(svc *dynamodb.DynamoDB) {
...
+          default:
+              fmt.Println(aerr.Error())
+          }
+      } else {
+          fmt.Println(err.Error())
+      }
+      return
+  }
+
+  type Manufacturers struct {
```
Comment (bgood) [inline examples/golang/main.go:474]: Let's move these to types.go

File: examples/golang/main.go [hunk 2]
```diff
@@ -229,13 +249,434 @@ func getProductsByCategory(svc *dynamodb.DynamoDB) {
...
+
+  fmt.Println("Verifying the customer has been deleted")
+  getCustomerContactDetails(svc,"CUST#9922885566","EMAIL#mosby@gmail.com")
+
+}
+
+func getCustomerWithSameId(svc *dynamodb.DynamoDB) {
+
+  fmt.Println("Running a scan operation to find customers with Id 0000000000 who has number_of_items greater than 3")
+  type Item struct {
```
Comment (bgood) [inline examples/golang/main.go:351]: Why not use the existing types or make a new `CustomerSummary` type or something like that.

**Review by bgood — CHANGES_REQUESTED** [review]
Couple comments on reusing types, but also a nit on indentation and stray white space.

**Review by gushob21 — COMMENTED** [review]
@bgood made changes as suggested. Please review them.

**Review by bgood — APPROVED** [review]

**Comment by bgood** [comment]
LGTM

Thank you!

**Pull request merged**